README.md

```diff
@@ -1,107 +1,196 @@
 # Express Routing
 
+When we go to a website like Facebook or Twitter, we see different pages for different URLs. In this lesson we are going to learn how add URLs to our Express app. We'll also look at some advanced techniques for how to manage routing once we have lot of URLs.
+
 ## Objectives
 
 1. Explain how to declare routes in Express
 2. Explain how callbacks are used to handle requests
 3. Describe how to modularize an application using `express.Router()`
 4. Explain how to mount a router on an Express application
 
-## Introduction
+## Structure of URLs
 
-Let's say that we have streets, and every street has buildings
+Before building our Express app, let's quickly review the parts of a URL. If we have a url such as `https://flatironschool.com/campuses/online/`, it is made up of the following parts:
 
-Streets are numbered consecutively
+* Protocol - `https`
+* Host - `flatiron.com`
+* Path - `/campuses/online`
 
+In this lesson we will be focusing on the `path` of the url, and how our app can do different things based on what the `path` is.
+
+## Hello Routes
+
+Let's start with our standard Hello World Express app:
+
+```js
+var express = require('express');
+var app = express();
+
+app.get('/', function(req, res) {
+  res.send('Hello World!');
+});
+
+app.listen(3000, function() {
+  console.log('Express app listening on port 3000!');
+});
 ```
-/streets[/{1-n}]
-```
 
-As are buildings:
+Visiting localhost:3000 in our browser, we should see `Hello World!`. Note that this is because our Express app treats, `localhost:3000` the same as `localhost:3000/`.
+
+Another way to test this route is to run `curl localhost:3000` in your terminal. Try this to make sure you get `Hello World!`.
+
+## How Does Routing Work?
+
+Try running `curl localhost:3000/hello`. You should see `Cannot GET /hello`. Why? Because our app doesn't know to handle the `/hello` path. Each time a request to our app is made, the Express app checks routes that have been registered.
 
+A route is a path and a callback function. We used `app.get` above to register a route. When a request is made for that path, the callback function that was the second argument to `app.get` gets called. That function is given a request and a response. In our callback function, we've called these arguments `req` (for request) and `res` (for response). Check out the docs to learn more about Express [requests](https://expressjs.com/en/4x/api.html#req) and [responses](https://expressjs.com/en/4x/api.html#res). We use the response argument's `send` method to send data back to the requester.
```

Comment:
Should we preface this with "In Express, a route is a path..."? While the pattern is basically the same anywhere (at least off the top of my head), I worry about students getting too hung up on, e.g., callbacks and their signatures in Express vs. some other web framework.